source/1_engine/rex_engine/src/diagnostics/logging/logger_config.cpp

```diff
@@ -28,81 +32,74 @@ namespace rex
+          {
+            end = i;
 
-      // return (name,value) trimmed pair from given "name=value" string.
-      // return empty string on missing parts
-      // "key=val" => ("key", "val")
-      // " key  =  val " => ("key", "val")
-      // "key=" => ("key", "")
-      // "val" => ("", "val")
+            if(start == end)
+            {
+              ++start;
+              continue;
+            }
 
-      inline rsl::pair<rex::DebugString, rex::DebugString> extract_kv_impl(char sep, const rex::DebugString& str)
-      {
-        auto n = str.find(sep);
-        rex::DebugString k;
-        rex::DebugString v;
-        if(n == rex::DebugString::npos())
-        {
-          v = str;
+            const rsl::string_view value = str.substr(start, end - start);
+            result.push_back(value);
+            start = end + 1;
+          }
         }
-        else
+
+        if(start != end)
         {
-          k = rex::DebugString(str.substr(0, n));
-          v = rex::DebugString(str.substr(n + 1));
+          const rsl::string_view value = str.substr(start, end - start);
+          result.push_back(value);
         }
-        return rsl::make_pair(trim_impl(k), trim_impl(v));
+
+        return result;
       }
 
       // return vector of key/value pairs from sequence of "K1=V1,K2=V2,.."
       // "a=AAA,b=BBB,c=CCC,.." => {("a","AAA"),("b","BBB"),("c", "CCC"),...}
-      inline rex::DebugHashTable<rex::DebugString, rex::DebugString> extract_key_vals_impl(const rex::DebugString& str)
+      rsl::unordered_map<rsl::string_view, rsl::string_view> extract_key_vals(rsl::string_view str)
       {
-        rex::DebugString token;
-        rex::DebugStringStream token_stream(str);
-        rex::DebugHashTable<rex::DebugString, rex::DebugString> rv {};
-        while(rsl::getline(token_stream, token, ','))
+        rsl::unordered_map<rsl::string_view, rsl::string_view> rv {};
+        const rsl::vector<rsl::string_view> key_value_pairs = split(str, ",");
+
+        for(const rsl::string_view key_value_pair: key_value_pairs)
         {
-          if(token.empty())
-          {
-            continue;
-          }
-          auto kv      = extract_kv_impl('=', token);
-          rv[kv.first] = kv.second;
+          rsl::vector<rsl::string_view> key_value = split(key_value_pair, "=");
+
+          REX_ASSERT_X(key_value.size() == 2, "Invalid logger level found at {}", key_value_pair);
+          rv[key_value[0]] = key_value[1];
         }
+
         return rv;
       }
 
-      void load_levels(const rex::DebugString& input)
+      void load_levels(rsl::string_view input)
       {
         if(input.empty() || input.size() > 512)
         {
           return;
         }
 
-        auto key_vals = extract_key_vals_impl(input);
-        rex::DebugHashTable<rex::DebugString, rexlog::level::LevelEnum> levels;
+        auto key_vals = extract_key_vals(input);
+        rsl::unordered_map<rsl::string_view, rexlog::level::LevelEnum> levels;
```

Comment:
You made a "using" for this, we should use it

Reply:
I made using for one using a debug allocator, this one shouldn't

Reply:
I cannot check this on my phone, but IIRC there is a using for this somewhere, could you double check pls

Reply:
fixed it